File: server/model/response/geo.py
"""地理与天气 API 响应模型。"""

from __future__ import annotations

from pydantic import BaseModel


class GeoMeResponse(BaseModel):
    ip: str # 公网 IP
    province: str | None = None # 省份
    city: str | None = None # 城市
    adcode: str | None = None # 行政区划代码
    weather: str | None = None # 天气
    temperature: str | None = None # 温度

File: web/src/api/geo.ts
import { authFetch } from "./client";

export interface GeoMeResponse {
  ip: string;
  province?: string | null;
  city?: string | null;
  adcode?: string | null;
  weather?: string | null;
  temperature?: string | null;
}

export async function getGeoMe(): Promise<GeoMeResponse | null> {
  try {
    const response = await authFetch("/geo/me");
    if (!response.ok) {
      return null;
    }
    return (await response.json()) as GeoMeResponse;
  } catch {
    return null;
  }
}
